### ISSUE TITLE
seeded topics words are not getting importance

### ISSUE THREAD

**Issue by ImSajeed** [vi3k6i5/GuidedLDA#31]
Hi @vi3k6i5 ,

I'm trying guided lda on six reviews data by initializing seed confiedence of 0.15, but they are not moving up the list as expected.

code below:

corpus =["Bus time arrive late and departure very late, arrived destination to late.","Bus is 1hr late at the boarding point and nearly 3hrs late at drop.","The bus arrived 1hour 45 mints late at the pickup point and was late by 2hrs 30min at destination.","Bus arrival time and departure time is not according to show in time schedule.","Bus timing is not correctly mentioned, it taken almost 6 hours not as mentioned.","Duration is mentioned as 8 hours 45 minutes, the travel timing took is 11 hours. Please change the duration to correct timings."]

df = pd.DataFrame(corpus,columns=['Review'])

import spacy

# Load English tokenizer, tagger, parser, NER and word vectors
nlp = spacy.load("en_core_web_sm")

from spacy.lang.en.stop_words import STOP_WORDS
from spacy.lang.en import English
import string
from unidecode import unidecode
import unicodedata

punctuations = string.punctuation
stopwords = list(STOP_WORDS)

stopwords = set(stopwords)-{'not','on'}

parser = English()
def spacy_tokenizer(sentence):
    mytokens = parser(sentence)
    mytokens = [ word.lemma_.lower().strip() if (word.lemma_ != "-PRON-" or word.lemma_ != "-X-") else word.lower_ for word in mytokens ]
    mytokens = [ word for word in mytokens if word not in stopwords and word not in punctuations ]
    #mytokens = remPunct(remNumbers(remNonAscii(mytokens)))
    mytokens = " ".join([i for i in mytokens])
    return mytokens

from tqdm import tqdm
tqdm.pandas()
df["cleaned_review"] = df['Review'].progress_apply(spacy_tokenizer)

all_review_list = [ review.split(' ') for review in df['cleaned_review']]

import nltk
from nltk.corpus import stopwords
from nltk.stem import PorterStemmer
from nltk.tokenize import WordPunctTokenizer
from nltk.collocations import BigramCollocationFinder
from nltk.metrics import BigramAssocMeasures

def get_bigrams(tokens):
    bigram_finder = BigramCollocationFinder.from_words(tokens)
    bigrams = bigram_finder.nbest(BigramAssocMeasures.chi_sq, 500)

    for bigram_tuple in bigrams:
        x = ' '.join(bigram_tuple)
        tokens.append(x)

    return tokens

import gensim

def bigrams(words, bi_min=10, tri_min=10):
    bigram = gensim.models.Phrases(words, min_count = bi_min)
    bigram_mod = gensim.models.phrases.Phraser(bigram)
    return bigram_mod
    
def get_corpus(words):
    bigram_mod = bigrams(words)
    bigram = [bigram_mod[review] for review in words]
    final_bigram = []
    for gram in bigram:
        try:
            d = get_bigrams(gram)
        except:
            final_bigram.append(gram)
            continue
        final_bigram.append(d)

    filtered_bigram = [[j for j in i if (j not in ['not','only','on'] and j.isdigit() == False)] for i in final_bigram]
    id2word = gensim.corpora.Dictionary(final_bigram)
    #id2word.filter_extremes(no_below=10, no_above=0.35)
    id2word.compactify()
    corpus = [id2word.doc2bow(text) for text in final_bigram]
    return corpus, id2word, filtered_bigram  

train_corpus, train_id2word, bigram_train = get_corpus(all_review_list)

vocab = []
for i in range(len(train_id2word)):
    vocab.append(train_id2word[i])

import numpy as np
from gensim import matutils
from gensim.matutils import corpus2csc

def bow_iterator(docs, dictionary):
    for doc in docs:
        yield dictionary.doc2bow(doc)

def get_term_matrix(msgs, dictionary):
    bow = bow_iterator(msgs, dictionary)
    X = np.transpose(matutils.corpus2csc(bow).astype(np.int64))
    return X

X = get_term_matrix(bigram_train, train_id2word)

import guidedlda
model = guidedlda.GuidedLDA(n_topics=2, n_iter=100, random_state=42, refresh=20)
model.fit(X)

topic_word = model.topic_word_
n_top_words = 20
for i, topic_dist in enumerate(topic_word):
    topic_words = np.array(vocab)[np.argsort(topic_dist)][:-(n_top_words+1):-1]
    print('Topic {}: {}'.format(i, ','.join(topic_words)))

Topic 0: late,arrive,bus,destination,point,pickup,pickup point,mint late,mint,late pickup,point nearly,bus arrive,arrive 1hour,45 mint,30min destination,30min,2hrs 30min,2hrs,1hour 45,1hour
Topic 1: time,mention,hour,time not,bus,bus time,departure,duration,accord time,mention 6,hour not,correctly,6 hour,time schedule,schedule,time departure,not mention,not accord,departure time,bus arrival

seed_topic_list = [['late pickup','point nearly','arrive 1hour','30min destination'],['accord time','hour not','time schedule'] ]

model = guidedlda.GuidedLDA(n_topics=2, n_iter=100, random_state=7, refresh=20)

word2id = dict((v, idx) for idx, v in enumerate(vocab))

seed_topics = {}
for t_id, st in enumerate(seed_topic_list):
    for word in st:
        seed_topics[word2id[word]] = t_id

model.fit(X, seed_topics=seed_topics, seed_confidence=0.15)

topic_word = model.topic_word_
n_top_words = 20
for i, topic_dist in enumerate(topic_word):
    topic_words = np.array(vocab)[np.argsort(topic_dist)][:-(n_top_words+1):-1]
    print('Topic {}: {}'.format(i, ','.join(topic_words)))

Topic 0: late,arrive,bus,destination,point,mint late,30min destination,point nearly,1hour,1hour 45,2hrs,2hrs 30min,30min,45 mint,pickup,arrive 1hour,nearly 3hrs,late 2hrs,late pickup,pickup point
Topic 1: time,bus,mention,hour,time not,departure,bus time,duration,accord time,mention 6,hour not,correctly,6 hour,time schedule,schedule,time departure,not mention,not accord,departure time,bus arrival
